docs/grover.rst

```diff
@@ -4,7 +4,57 @@ Grover's Search Algorithm and Amplitude Amplification
 Overview
 --------
 
-Quantum search algorithm and more via amplitude amplification.
+This module implements Grover's Search Algorithm, and the more general Amplitude Amplification
+Algorithm. Grover's Algorithm solves the following problem:
+
+Given a collection of basis states {:math:`\ket{y}_i`}, and a quantum circuit that assigns these
+states a common phase unique from the other basis states, construct a quantum circuit that when
+given the uniform superposition :math:`\ket{s} = \frac{1}{\sqrt{N}}\sum\limits^{N-1}_{i=0}\ket{x_i}`
+as input, will produce a state :math:`\ket{s'}` with nearly all of its support on the states
+:math:`\{\ket{y}_i\}`.
```

Comment:
This sentence is highly convoluted. Can you give an example to clarify the problem statement?

Reply:
Yes!